OpenSim/Simulation/Model/PathPoint.h

```diff
@@ -35,88 +35,76 @@ class PhysicalFrame;
-#endif
-    SimTK::Vec3& getLocation()  { return upd_location(); }
-
-    const double& getLocationCoord(int aXYZ) const {
-        assert(aXYZ>=0 && aXYZ<=2); return get_location()[aXYZ];
-    }
-
-    // A variant that uses basic types for use by GUI
-    void setLocationCoord(int aXYZ, double aValue) {
+    // A variant that uses basic types for use by GUI, e.g. OpenSimContext
+    /** <b>(Deprecated)</b> Old PathPoint interface */
+    DEPRECATED_14("Use setLocation() instead.")
+    void setLocationCoord(const SimTK::State&s, int aXYZ, double aValue) {
         assert(aXYZ>=0 && aXYZ<=2);
-        upd_location()[aXYZ]=aValue;
+        updStation().upd_location()[aXYZ]=aValue;
     }
 
-    void setLocation( const SimTK::State& s, const SimTK::Vec3& location) {
-        // TODO this is completely wrong to be taking a state and changing a property!
-        upd_location() = location;
-    }
-    void setLocation( const SimTK::State& s, int aCoordIndex, double aLocation) {
-        if (aCoordIndex >= 0 && aCoordIndex <= 2)
-           upd_location()[aCoordIndex] = aLocation;
-    }
-#ifndef SWIG
-    // A variant that uses basic types for use by GUI
-    void setLocation( const SimTK::State& s, double pt[]){ 
-        setLocation(s,SimTK::Vec3::updAs(pt));
-    }
-#endif
-    void setBody(const PhysicalFrame& body) {
-        this->setParentFrame(body);
+    SimTK::Vec3 getLocation(const SimTK::State& s) const override {
+        return getStation().get_location();
     }
 
+    void setLocation(const SimTK::Vec3& location);
+
     void changeBodyPreserveLocation(const SimTK::State& s,
                                     const PhysicalFrame& body);
 
-    const PhysicalFrame& getBody() const { return getParentFrame(); }
-    const std::string& getBodyName() const { return getParentFrame().getName(); }
+    void scale(const SimTK::Vec3& scaleFactors) override;
 
-    virtual void scale(const SimTK::Vec3& scaleFactors) {
-        for (int i = 0; i < 3; i++)
-            upd_location()[i] *= scaleFactors[i];
-    }
+protected:
 
-    virtual const WrapObject* getWrapObject() const { return NULL; }
+    const Station& getStation() const { 
+        return getMemberSubcomponent<Station>(stationIx);
+    }
+    Station& updStation() {
+        return updMemberSubcomponent<Station>(stationIx);
+    }
 
-    virtual bool isActive(const SimTK::State& s) const { return true; }
+    // Component Interface
+    void extendFinalizeFromProperties() override;
+    void extendConnectToModel(Model& model) override;
 
-    // get the relative velocity of the path point with respect to the body
-    // it is connected to.
-    virtual void getVelocity(const SimTK::State& s, SimTK::Vec3& velocity) {
-        velocity.setToZero();
+private: 
+    // Satisfy Point interface
+    /* Calculate the location of this PathPoint in Ground as a function of
+       the state. */
+    SimTK::Vec3 calcLocationInGround(const SimTK::State& state) const final {
```

Comment:
I think they are implementing virtual functions (thus the `final`), and the virtual function assumes a calculation may be necessary. However, we decided to mark a final function as `override final`, not simply `final` (we went back and forth on this, but ultimately decided on `override final`: https://github.com/opensim-org/opensim-core/pull/299).